recipes/cudatoolkit/build.py

```diff
@@ -0,0 +1,405 @@
+        self.nvtoolsext_fmt = "{0}64_1.dll"
+        self.libdevice_lib_fmt = "libdevice.10.bc"
+        pfs = ["Program Files", "Program Files (x86)"]
+        nvidias = ["NVIDIA Corporation", "NVIDIA GPU Computing Toolkit"]
+        nvtxs = ["NVToolsExt", "NvToolsExt", "nvToolsExt"]
+        for pf, nvidia, nvtx in itertools.product(pfs, nvidias, nvtxs):
+            nvt_path = os.path.join("c:" + os.sep, pf, nvidia, nvtx, "bin")
+            if os.path.exists(nvt_path):
+                self.nvtoolsextpath = nvt_path
+                break
+        else:
+            self.nvtoolsextpath = None
+        self.libdir = "Library/bin"
+        self.post_init()
+
+    def copy(self, *args):
+        (store,) = args
+        self.copy_files(cuda_lib_dir=store, nvvm_lib_dir=store, libdevice_lib_dir=store)
+
+    def extract(self):
+        try:
+            with tempdir() as tmpd:
+                extract_name = "__extracted"
+                extractdir = os.path.join(tmpd, extract_name)
+                os.mkdir(extract_name)
+
+                check_call(
+                    [
+                        "7za",
+                        "x",
+                        "-o%s" % extractdir,
+                        os.path.join(self.src_dir, self.runfile),
+                    ]
+                )
+                for p in self.patches:
+                    check_call(
+                        [
+                            "7za",
+                            "x",
+                            "-aoa",
+                            "-o%s" % extractdir,
+                            os.path.join(self.src_dir, p),
+                        ]
+                    )
+
+                nvt_path = os.environ.get(
+                    "NVTOOLSEXT_INSTALL_PATH", self.nvtoolsextpath
+                )
+                print("NvToolsExt path: %s" % nvt_path)
+                if nvt_path is not None:
+                    if not Path(nvt_path).is_dir():
+                        msg = "NVTOOLSEXT_INSTALL_PATH is invalid " "or inaccessible."
+                        raise ValueError(msg)
+
+                # fetch all the dlls into DLLs
+                store_name = "DLLs"
+                store = os.path.join(tmpd, store_name)
+                os.mkdir(store)
+                for path, dirs, files in os.walk(extractdir):
+                    if (
+                        "jre" not in path and "GFExperience" not in path
+                    ):  # don't get jre or GFExperience dlls
+                        for filename in fnmatch.filter(files, "*.dll"):
+                            if not Path(os.path.join(store, filename)).is_file():
+                                shutil.copy(os.path.join(path, filename), store)
+                        for filename in fnmatch.filter(files, "*.lib"):
+                            if not Path(os.path.join(store, filename)).is_file():
+                                shutil.copy(os.path.join(path, filename), store)
```

Comment:
At present this recipe has the same issue as https://github.com/AnacondaRecipes/cudatoolkit-feedstock/issues/12 - see also the PR to fix it at https://github.com/AnacondaRecipes/cudatoolkit-feedstock/pull/14. The change that would fix the issue here is:

```suggestion
                            if path.endswith('x64'):
                                if not Path(os.path.join(store, filename)).is_file():
                                    shutil.copy(os.path.join(path, filename), store)
```

Reply:
Thanks @gmarkall!